src/loaders/helpers/apply-middlewares.ts

```diff
@@ -54,13 +54,13 @@ export function applyAfterAuthMiddleware(app: Router, middlewares: GetInjectable
 }
 
 function applyMiddleware(app: Router, middleware: GetInjectableOption<'middleware'>): void {
-	app.use(async (req: MedusaRequest | MedusaAuthenticatedRequest, res: Response, next: NextFunction) => {
-		const shouldHandle = middleware.routes.some((route) => {
-			return (
-				(route.method === 'all' || req.method.toLowerCase() === route.method.toLowerCase()) &&
-				(route.path === '*' || req.originalUrl === route.path)
-			);
-		});
+	const routes = middleware.routes.map((routeObj) => routeObj.path);
+	app.use(routes, async (req: MedusaRequest | MedusaAuthenticatedRequest, res: Response, next: NextFunction) => {
+		const shouldHandle = middleware.routes
+			.filter((route) => route.path === req.originalUrl)
```

Comment:
Actually this code is incorrect. route.path === req.originalUrl will false in case of parametric routes. Need to come up with a different solution